File: Tune/app/src/main/java/com/example/tune/activities/adapters/CustomPreferencesAdapter.java
package com.example.tune.activities.adapters;

import android.app.Activity;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.example.tune.R;
import com.example.tune.activities.PostActivity;
import com.example.tune.activities.models.Genre;

public class CustomPreferencesAdapter extends ArrayAdapter {

    private Genre[] genres;
    private Activity context;
    private Genre[] preferences;

    public CustomPreferencesAdapter(Activity context, Genre[] genres, Genre[] preferences) {
        super(context, R.layout.timeline_row, genres);
        this.genres = genres;
        this.context = context;
        this.preferences = preferences;
    }

    public View getView(final int position, View view, ViewGroup parent){
        LayoutInflater inflater = context.getLayoutInflater();
        View rowView = inflater.inflate(R.layout.preferences_row, null, true);

        TextView genreTxt = (TextView) rowView.findViewById(R.id.preferencesTxt);

        genreTxt.setText(genres[position].toString());

        RadioButton radioButton = rowView.findViewById(R.id.radioBtnPreferences);

        for (Genre genre: preferences) {
            if(genres[position] == genre){
                radioButton.setChecked(true);
            }
        }

        if(position % 2 == 1){
            rowView.setBackgroundColor(ContextCompat.getColor(context, R.color.lightBlue));
        } else {
            rowView.setBackgroundColor(ContextCompat.getColor(context, R.color.darkBlue));
        }

        return rowView;
    }
}

File: tune_kotlin/app/src/main/java/com/example/tune_kotlin/adapters/CustomPreferencesAdapter.kt
package com.example.tune_kotlin.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import com.example.tune_kotlin.R
import com.example.tune_kotlin.models.Genre

class CustomPreferencesAdapter(private val context: Context, private val preferences: ArrayList<String>) : BaseAdapter() {

    private val genres = Genre.values()

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view: View?
        val viewHolder: ViewHolder
        val inflater: LayoutInflater
                = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        if(convertView == null){
            view = inflater.inflate(R.layout.preferences_row, parent, false)
            viewHolder = ViewHolder()
            viewHolder.genre = view.findViewById(R.id.preferencesTxt)
            viewHolder.checkbox = view.findViewById(R.id.checkboxPreferences)

            view.tag = viewHolder
        } else {
            view = convertView
            viewHolder = convertView.tag as ViewHolder
        }

        val genreTxt = viewHolder.genre
        val checkBox = viewHolder.checkbox

        preferences.map { genre ->
            if(genre == genres[position].name){
                checkBox.isChecked = true
            }
        }

        checkBox.setOnClickListener {
            checkBox.isChecked = !checkBox.isChecked
        }
        return view!!
    }

    override fun getItem(position: Int): Any {
        return genres[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return genres.size
    }

    private class ViewHolder {
        lateinit var genre: TextView
        lateinit var checkbox: CheckBox
    }
}